cmake/list_append_keys_from.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
+#
+macro(list_append_keys_from listname source)
```

Comment:
I didn't want to make this macro overly complex and spared the outer loop for `ARGN` since it was not needed for the use case. Its calling conventions differs from `list_append_unique`, because it simplified the calling code.

I am not sure if this can be implemented as a function. I have not found any way to read a named variable that was defined in the parent scope.